api/API_VERSIONING.md

```diff
@@ -0,0 +1,198 @@
+  procedure.
+
+* Renaming of fields or package namespaces for a proto must not occur. This is inherently dangerous,
+  since:
+  * Field renames break wire compatibility. This is stricter than standard proto development
+    procedure in the sense that it does not break binary wire format. However, it **does** break
+    loading of YAML/JSON into protos as well as text protos. Since we consider YAML/JSON to be first
+    class inputs, we must not change field names.
+
+  * For service definitions, the gRPC endpoint URL is inferred from package namespace, so this will
+    break client/server communication.
+
+  * For a message embedded in an `Any` object, the type URL, which the package namespace is a part
+    of, may be used by Envoy or other API consuming code. Currently, this applies to the top-level
+    resources embedded in `DiscoveryResponse` objects, e.g. `Cluster`, `Listener`, etc.
+
+  * Consuming code will break and require source code changes to match the API changes.
+
+* Some other changes are considered breaking for Envoy APIs that are usually considered safe in
+  terms of protobuf wire compatibility:
+  * Upgrading a singleton field to a repeated, e.g. `uint32 foo = 1;` to `repeated uint32 foo = 1`.
+    This changes the JSON wire representation and hence is considered a breaking change.
+
+  * Wrapping an existing field with `oneof`. This has no protobuf or JSON/YAML wire implications,
+    but is disruptive to various consuming stubs in languages such as Go, creating unnecessary
+    churn.
+
+  * Increasing the strictness of
+    [protoc-gen-validate](https://github.com/envoyproxy/protoc-gen-validate) annotations. Exceptions
+    may be granted for scenarios in which these stricter conditions model behavior already implied
+    structurally or by documentation.
+
+The exception to the above policy is for API versions tagged `vNalpha`. Within an alpha major
+version, arbitrary breaking changes are allowed.
+
+Note that changes to default values for wrapped types, e.g. `google.protobuf.UInt32Value` are not
+governed by the above policy. Any management server requiring stability across Envoy API or
+implementations within a major version should set explicit values for these fields.
+
+# API lifecycle
+
+The API lifecycle follows a calendar clock. At the end of Q3 each year, a major API version
+increment may occur for any Envoy API package, in concert with the quarterly Envoy release.
+
+Envoy will support at most three major versions of any API package at all times:
+* The current stable major version, e.g. v3.
+* The previous stable major version, e.g. v2. This is needed to ensure that we provide at least 1
+  year for a supported major version to sunset. By supporting two stable major versions
+  simultaneously, this makes it easier to coordinate control plane and Envoy rollouts as well. This
+  previous stable major version will be supported for at most 1 year.
```

Comment:
"...for **at least** (?) 1 year **after introduction of the following version** (?)"